### PR TITLE
Add support for SimpleTask, Deadline and Event

### LINKED ISSUE #55: Create Todo Class
Create Todo class to replace Task which is to be made abstract.

### PR DESCRIPTION
**Pull request by Huggenguggen**
Promote Task to abstract class.
Add 3 subclasses SimpleTask, Deadline, and Event.
Add support in addCommandParser to add different type of tasks dynamically.
Change representation in JsonAdaptedTask such that it can represent new types of task and still parse them.

Update and create relevant test cases and documentation.

closes #55 #56 #57 #58

### CODE REVIEW

File: src/test/java/seedu/task/storage/JsonAdaptedTaskTest.java [hunk 1]
```diff
@@ -35,39 +40,73 @@ public void toModelType_validTaskDetails_returnsTask() throws Exception {
...
+        assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
+    }
+
+    @Test
+    public void toModelType_deadlineAndEventOverlap_throwsIllegalValueException() {
+        JsonAdaptedTask task =
+                new JsonAdaptedTask(VALID_NAME, VALID_DESCRIPTION, VALID_TAGS, VALID_DATE, VALID_DATE, VALID_DATE);
+        String expectedMessage = DEADLINE_EVENT_OVERLAP;
+        assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
+    }
```
Comment (lywich) [inline src/test/java/seedu/task/storage/JsonAdaptedTaskTest.java:102]: Nice work. Since we are already testing this case, how about also testing situations where by the JSON date format is  (EMPTY, VALID, EMPTY) and (EMPTY, EMPTY, VALID).

**Review by lywich — COMMENTED** [review]
Great work as always, I have left some suggestions for the testcases to test for some edge cases which I hope could improve upon that section.

**Review by lywich — APPROVED** [review]
Alright nice work. Well done on the quick changes.
